src/test/java/de/intension/keycloak/KeycloakExtensionTestBase.java

```diff
@@ -0,0 +1,147 @@
+                if ((re.getCause() != null) && (re.getCause() instanceof java.net.BindException)) {
+//                    logger.debug("Setup failed for SMTP Server before test run: " + re.getCause().getMessage());
+                    retryCount--;
+                }
+                else {
+                    throw re;
+                }
+            }
+        }
+
+    }
+
+    protected String getAdminToken()
+        throws Exception
+    {
+        HttpClient client = HttpClient.newHttpClient();
+        Map<String, String> params = new HashMap<>();
+        params.put("grant_type", "password");
+        params.put("client_id", "admin-cli");
+        params.put("username", keycloak.getAdminUsername());
+        params.put("password", keycloak.getAdminPassword());
+        HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create("http://localhost:" + keycloak.getHttpPort() + "/auth/realms/" + REALM
+                    + "/protocol/openid-connect/token"))
+            .header("Content-Type", "application/x-www-form-urlencoded").POST(buildFormDataFromMap(params)).build();
+        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        return retrieveToken(response.body());
+    }
+
+    protected HttpResponse<String> post(String url, String body)
+        throws Exception
+    {
+        HttpClient javaClient = HttpClient.newHttpClient();
+        HttpRequest request = HttpRequest.newBuilder().uri(URI.create(url))
+            .header("Authorization", "Bearer " + getAdminToken())
+            .header("Content-Type", "application/json").POST(HttpRequest.BodyPublishers.ofString(body)).build();
+        return javaClient.send(request, HttpResponse.BodyHandlers.ofString());
+    }
+
+    private void startServer()
+    {
+        smtpServer = new Wiser();
+        smtpServer.setHostname("localhost");
+        smtpServer.setPort(25);
+        smtpServer.start();
+        smtpServer.getMessages().clear();
+    }
+
+    private String retrieveToken(String responseBody)
+        throws Exception
+    {
+        return mapper.readValue(responseBody, AccessTokenResponse.class).getToken();
+    }
+
+    private HttpRequest.BodyPublisher buildFormDataFromMap(Map<String, String> params)
+    {
+        var builder = new StringBuilder();
+        for (Map.Entry<String, String> entry : params.entrySet()) {
+            if (builder.length() > 0) {
+                builder.append("&");
+            }
+            builder.append(URLEncoder.encode(entry.getKey().toString(), StandardCharsets.UTF_8));
+            builder.append("=");
+            builder.append(URLEncoder.encode(entry.getValue().toString(), StandardCharsets.UTF_8));
+        }
+        return HttpRequest.BodyPublishers.ofString(builder.toString());
+    }
+
+    @AfterEach
+    public void stopSMTPServer()
+    {
+        smtpServer.stop();
+        smtpServer = null;
+    }
+
+    public List<WiserMessage> getMessages()
```

Comment:
the Wiser SMTP server isn't used in the tests at all, right? So you can safely remove it and the dependency